extension/dynamicconfig/model/collection_period.go

```diff
@@ -19,20 +19,65 @@
 package model
 
 import (
-	pb "github.com/open-telemetry/opentelemetry-proto/gen/go/collector/dynamicconfig/v1"
+	"encoding/binary"
+	"fmt"
 	"hash/fnv"
+	"strconv"
 )
 
 type CollectionPeriod string
 
-func (period CollectionPeriod) Proto() pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod {
-	interval := pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod_value[string(period)]
-	return pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod(interval)
+func (period CollectionPeriod) Proto() (int32, error) {
```

Comment:
Let's add a TODO here to see if there is a library for parsing time duration.

1s = 1s
1m  = 60s
1h = 3600s

Reply:
Done